pkg/name/name.go

```diff
@@ -274,7 +273,7 @@ func getFromStructPath(node interface{}, path util.Path) (interface{}, bool, err
 	var structElems reflect.Value
 	switch kind {
 	case reflect.Map, reflect.Slice:
-		if len(path) != 0 {
+		if len(path) == 0 {
```

Comment:
This was indeed wrong but actually can be completely removed since there's a check at the start.